### ISSUE TITLE
Issues integrating into existing Microsoft SQL database

### ISSUE THREAD

**Issue by levipro** [zenstackhq/zenstack#724]
**Description and expected behavior**
We have an SQL database with many existing tables. I ran the following commands:

```
prisma db pull
zenstack init
```

Both of these had no issues. Next I ran:

`zenstack generate`

There were a few issues with this:

1. Invalid zmodel generated from prisma schema:

Attribute "@default" doesn't have a parameter named "map"
```
CreatedDate               DateTime                    @default(now(), map: "DF_Address_CreatedDate") @db.DateTime
```

Value is not assignable to parameter [Max]
```
NVarChar(Max)
VarChar(Max)
```

So to resolve these I manually removed `, map: "DF_Address_CreatedDate"` and then replaced `Max` with a number just to test.

2. After that I received the error `Maximum call stack size exceeded`

To get this to work I had to pass `--stack-size=2048 --max-old-space-size=12288` to node 

**Environment (please complete the following information):**

-   ZenStack version: 1.0.0-beta.24
-   Prisma version: 5.3.1
-   Database type: Microsoft SQL

**Additional Details**

I have enabled the trpc plugin
```
plugin trpc {
  provider = '@zenstackhq/trpc'
  output = 'src/server/routers/generated'
}
```

**Comment by ymc9** [comment]
Hi @levipro , thanks for reporting this! Currently `@default` doesn't support the second argument yet. I think you can workaround using the `@prisma.passthrough` attribute:

```prisma
CreatedDate               DateTime                    @prisma.passthrough('@default(now(), map: "DF_Address_CreatedDate")') @db.DateTime
```

**Comment by levipro** [comment]
Thanks @ymc9 is there any work around for 

NVarChar(Max)
VarChar(Max)

**Comment by ymc9** [comment]
> Thanks @ymc9 is there any work around for
> 
> NVarChar(Max) VarChar(Max)

Sorry I missed that part. I'm actually not sure where this error came from. Do you mind sharing the zmodel snippet that caused it? Thanks! @levipro

**Comment by levipro** [comment]
Here is an example repo
https://github.com/levipro/zenstack-issue

```
pnpm zenstack generate
Validation errors:
line 20: Could not resolve reference to ReferenceTarget named 'Max'. [Max]
line 21: Could not resolve reference to ReferenceTarget named 'Max'. [Max]
line 20: Value is not assignable to parameter [Max]
line 21: Value is not assignable to parameter [Max]
line 20: expression cannot be resolved [Max]
line 21: expression cannot be resolved [Max]
```

**Comment by ymc9** [comment]
I see. You can use the same "@prisma.passthrough" attribute to workaround. It basically passes text directly to the generated prisma schema:

```prisma
Latitude                 String? @prisma.passthrough('@db.VarChar(Max)')
```

**Comment by ymc9** [comment]
Hi @levipro , I'm working on fixing this bug couldn't find Prisma's documentation for the `@db.VarChar(Max)` definition (for the `Max` enum). Just wondering you may know where it's documented. I want to make sure to add other missing definitions as well.

**Comment by levipro** [comment]
I don't have much information on that, maybe these might be a little help:
https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference
https://www.prisma.io/dataguide/datamodeling/tables-tuples-types

**Comment by ymc9** [comment]
Thanks! Finally found something in a link to Microsoft's doc. It seems `Max` is the only missing thing for that part.

**Comment by ymc9** [comment]
Hi @levipro , both issues are fixed in v1.1.0
